### DISCUSSION TITLE
Question about the x86_64 branch!

### DISCUSSION THREAD

**Discussion by lloydbayley** [Nachtzuster/BirdNET-Pi#13] [General]
Quick question for @Nachtzuster : Is the x86_64 branch being actively maintained (as far as you can manage, I know you're just tidying things up) along with the main branch? If so, I'd like to update my VM to that branch. However, the install files are still reflecting the main branch in that x86_64 branch. I could update it but it would be overwritten in the next update. I understand that it may stay static after this but was wondering if it could be used live yet?

**Comment by Nachtzuster** [comment]
I'm using that x86_64 branch for development of other _actual_ 'feature' branches: so I rebase it onto main whenever I need it for testing. (way quicker and easier than messing with sd cards an RPi's :wink:) Using it for 'production' is therefore not advised.
Merging to main would ease my workflow, but reasons for not merging with to main are
- Fonts look funny in `View Log` and `Web Terminal` due to the updated `gotty` - not sure how to fix this.
- I don't want to advertise x86_64 support - feels like opening Pandora's box (could mark it experimental maybe)

**Reply by lloydbayley** [reply]
Completely understood! Just thought I'd ask.
That's weird about the fonts. Gotty is a bit of a pain when it does things like that. It doesn't like change! :)
We don't want to release all the horrible things in the world from Pandora's Box!!!

**Comment by lloydbayley** [comment]
@Nachtzuster Just FYI, I have installed a test instance of your x86_64 branch (with a small branch-pointer modification) and it went in first time to a ProxMox debian bookworm VM without issue. Also, the Web Terminal and logs show up fine. It's only a play install but just thought the results might be interesting for you!

**Comment by Nachtzuster** [comment]
Thank for the confirmation!
Couple of questions:
Does gotty look ok on that install? (screenshot?)
Which browser are you using?

**Reply by lloydbayley** [reply]
Sorry, yes that's what I meant when I said that the Web Terminal and Logs looked fine. I remembered what you said about gotty and I tried it on Firefox, Chrome & Safari. All working fine. Will take screenshots in a min.

**Reply by lloydbayley** [reply]
Ok. Screenshots:

Chrome:
<img width="1403" alt="Chrome Screenshot - Nachtzuster" src="https://github.com/Nachtzuster/BirdNET-Pi/assets/21309308/0e1a6d27-d046-44ef-9f62-2badaa29f37c">

Safari:
<img width="1224" alt="Safari Screenshot - Nachtzuster" src="https://github.com/Nachtzuster/BirdNET-Pi/assets/21309308/f8d300be-b2ec-4f29-a172-c51deb426328">

Firefox:
<img width="1425" alt="Firefox Screenshot - Nachtzuster" src="https://github.com/Nachtzuster/BirdNET-Pi/assets/21309308/0dddbd07-a8d9-4a5e-83d4-bf333b62b89e">

**Reply by Nachtzuster** [reply]
That does look fine indeed. - I wonder why it looks so horrible in my Firefox on Ubuntu 
![image](https://github.com/Nachtzuster/BirdNET-Pi/assets/5863099/969887a4-1d17-4848-9ed6-12f97998be91)

(Chromium and mobile Firefox look fine here)

**Reply by lloydbayley** [reply]
I've no idea...I mean the screenshot looks readable albeit with some extra vertical space but it's a little 'jagged' as you point out. Could it be the font cache in your end needs refreshing?

**Comment by Nachtzuster** [comment]
There _is_ something wrong with your install though: I'm seeing NaN 'not a number' in your logs. the analysis is deeply unhappy :-)

**Reply by lloydbayley** [reply]
Yes, it's just a test install, there is no mic connected at this time, thus the NaN....I just flashed the VM up to get you the screenshots!

**Comment by Nachtzuster** [comment]
I decided to merge the x86_64 branch as it does simplify my workflow.

The font issue will only affect a tiny percentage of users, I think, especially as I'm keeping the aarch64 version on the old version.

**Reply by lloydbayley** [reply]
Oh right! Well as long as it's useful to you. Don't do it if it mucks things up.
In other news, I have had a mic hooked up to my test VM of the x86_64 branch and it picked up a comparable amount of birds to my live system. The mic is not in an ideal place but it showed me that things were fine. I also flicked the main MIC across to the VM for half an hour to test with the mic in the best place and that was good. So, I can't fault that branch at all on a Proxmox x86 VM!

**Reply by kniftar** [reply]
I just installed this on an HP-T620 thin client running bookworm/LXQT.  The install went ok but no detections/logs. I had to uninstall TFLite_runtime and install Tensorflow then everything is working perfectly as far as I can tell. Not sure if it will survive an update though.  Thanks so much for this - it was the easiest x86 install I've tried.

**Reply by Birdsphotos** [reply]
Could you provide more detail on the "I had to uninstall TFLite_runtime and install Tensorflow" as i am having the same no detections/ logs issue

**Reply by kniftar** [reply]
The instructions were posted on this thread:
https://github.com/Nachtzuster/BirdNET-Pi/discussions/111#discussioncomment-10229622
The TFlite runtime is for ARM processors so need replaced with x86 version. After that everything is working great.

**Reply by Nachtzuster** [reply]
this does not sound right: this discussion is about adding x86_64 to the installer. 
ie so that it installs the x86_64 version of TFLite runtime when it detects x86_64.

Could still be that the x86_64 version of TFLite runtime needs certain cpu instructions not available on certain older cpu's, or not passed to your vm/container.

It does run on my _ancient_ 4th gen intel laptop though.

**Reply by Birdsphotos** [reply]
I was able to get it to run when I installed it on Ubuntu. It was not working on Debian 12.7 even after I tried the instructions referred to in #111 above

**Reply by Nachtzuster** [reply]
Interesting... not sure why that would be without install and/or runtime logs though

**Reply by kniftar** [reply]
Well I could certainty be wrong about that I'm not super literate about this stuff.  I didn't know there WAS a x86_64 runtime.  On your readme under Introduction is says [[using [[pre-built TFLite binaries](url)]](https://github.com/PINTO0309/TensorflowLite-bin)
When I goto that link it only talks about arm and raspberry pi's. Didn't see any x86_64 version mentioned so that's why I thought the tflite binary was for arm.  When I installed tensorflow it looked like it compiled for my hardware and then worked.  
My install is not on a VM, it's on an HP-T620 thin client with an AMD G-Series GX-217GA SOC: dual-core APU
1.65GHz.

**Reply by kniftar** [reply]
> Interesting... not sure why that would be without install and/or runtime logs though

Here's my install log if it could help

**Reply by Nachtzuster** [reply]
Thanks!
Looks like the installer did the right thing and pulled the x86_64 version of TFLite. The rest also looks as expected, so all good here.
(Maybe remove the log now for your privacy)

Maybe the runtime log could still remove some of the mystery? If the logs are still on that machine: `journalctl -u birdnet_analysis.service` and scroll to the relevant date/time.

**Reply by kniftar** [reply]
Ok, here's the relevant runtime log snip:
[analysis.log](https://github.com/user-attachments/files/17298144/analysis.log)
Looks like it couldn't find the module then after I installed tenserflow and rebooted it starts working.  Maybe something ro do with the AMD processor?

**Reply by Nachtzuster** [reply]
Thanks.
Weird that it says it can't find the module. I was expecting something like 
`The TensorFlow library was compiled to use AVX2 instructions, but these aren't available on your machine.
Aborted` That is what I get on an emulated SandyBridge machine.

Anyway I guess the takeaway is that the TFLite [build](https://github.com/Nachtzuster/BirdNET-Pi/releases/tag/v0.1) this repo uses needs at least a Haswell (AVX2)

**Comment by lloydbayley** [comment]
Well, today I migrated my live install on a Proxmox VM to your (main) repo and all is working as it should. Just reporting in! :)
